zebra-state/src/sled_state.rs

```diff
@@ -116,10 +118,21 @@ impl FinalizedState {
                 block_by_height.insert(&height_bytes, block_bytes)?;
 
                 // for some reason type inference fails here
-                Ok::<_, sled::transaction::ConflictableTransactionError>(())
-            });
+                Ok::<_, sled::transaction::ConflictableTransactionError>(hash)
+            })
+            .map_err(Into::into)
+    }
 
-        let _ = rsp_tx.send(transaction_result.map(|_| hash).map_err(Into::into));
+    /// Commit a finalized block to the state.
+    ///
+    /// It's the caller's responsibility to ensure that blocks are committed in
+    /// order. This function is called by [`queue`], which ensures order.
+    /// It is intentionally not exposed as part of the public API of the
+    /// [`FinalizedState`].
+    pub fn commit_finalized(&mut self, queued_block: QueuedBlock) {
```

Comment:
Do we need to adjust the visibility of this function?

```suggestion
    /// It is intentionally not exposed as part of the public API of the
    /// [`FinalizedState`].
    fn commit_finalized(&mut self, queued_block: QueuedBlock) {
```